### PR TITLE
Fix errors in av1 decoder interfaces

### PR DESCRIPTION
**Pull request by rosetta-jpn**
Signed-off-by: Hirokazu Honda <hiroh@chromium.org>

### CODE REVIEW

File: va/va_dec_av1.h [hunk 1]
```diff
@@ -365,8 +365,8 @@ typedef struct  _VADecPictureParameterBufferAV1
     uint8_t                 tile_rows;
 
     /* specifes the width/height of a tile minus 1 in units of superblocks */
-    uint16_t                width_in_sbs_minus_1[63];
-    uint16_t                height_in_sbs_minus_1[63];
+    uint16_t                width_in_sbs_minus_1[64];
```
Comment (XinfengZhang) [inline va/va_dec_av1.h:368]: it will break backward compatibility

Comment (rosetta-jpn) [reply va/va_dec_av1.h:368]: Should I decrease this by 1? https://github.com/intel/libva/blob/master/va/va_dec_av1.h#L650

Comment (dvrogozh) [reply va/va_dec_av1.h:368]: This won't work and still breaks backward compatibility. You can not add fields or increase size of arrays in the middle of the structure. You need to add fields in the very end taking out space from va_reserved. 

Why exactly do we need this change?

**Comment by cwang105** [comment]
why do we want to change width_in_sbs_minus_1[63] to width_in_sbs_minus_1[64]?
The max number of tile columns is 64. So the width of last tile column can always be derived from frame width, no?

Thanks,
Ce

**Comment by rosetta-jpn** [comment]
It sounds good to me if the driver computes tile_width as sb columns from them.
It was not obvious that a driver needs to compute it from the comment.

**Comment by XinfengZhang** [comment]
@rosetta-jpn , to avoid backward compatible issue , change the comments is enough, right? will you change it?

**Comment by rosetta-jpn** [comment]
Sure. Updated.

**Comment by XinfengZhang** [comment]
@rosetta-jpn , could you help to squash two commits together.

**Comment by rosetta-jpn** [comment]
> @rosetta-jpn , could you help to squash two commits together.

Done!

**Comment by XinfengZhang** [comment]
LGTM, @cwang105 @dvrogozh @ZefuLI if there are no other concern, will merge it. thanks

**Comment by rosetta-jpn** [comment]
ping other reviewers.

By the way, I would like to ask you all if marking deprecated to tg_start and tg_end is a correct thing.
https://github.com/intel/libva/issues/431#issuecomment-652735631

**Comment by cwang105** [comment]
Looks good to me.

Thanks,
Ce

**Comment by XinfengZhang** [comment]
from my personal understanding, tg_start and tg_end should not be deprecated now. 
yes, https://github.com/intel/media-driver does not care about it at all, but the interface is designed not only for Intel. maybe some underlying driver need these parameters. I am not whether these parameter is independent and will not impact implementation.

**Comment by rosetta-jpn** [comment]
Sorry for late reply.

> from my personal understanding, tg_start and tg_end should not be deprecated now.
tg_start and tg_end were confusing for me at least.
Since there are tg_column and tg_row, looks like its structure should be used for a single tile.
But tg_start and tg_end are contradicts to it.

Yes, Deprecated might not be correct.
Shall we clarify this by some comment in the structure? Any suggestion about the comment?

**Comment by rosetta-jpn** [comment]
Gentle ping

**Comment by XinfengZhang** [comment]
@rosetta-jpn , sorry I missed the notification mail. I am fine with current change, if no other concern, I will merge it , thanks

**Pull request merged**